pkg/config/legacy/converter.go

```diff
@@ -283,6 +338,30 @@ func extractURLAPIKeys(agentConfig Config, configConverter *config.LegacyConfigC
 		}
 	}
 	configConverter.Set("additional_endpoints", additionalEndpoints)
+
+	// Process url + additional endpoints
+	if v, ok := agentConfig["process.config.endpoint"]; ok {
+		processEndpoints := map[string][]string{}
+
+		processURLs, err := splitString(v, ',')
+		if err != nil || len(processURLs) == 0 {
+			return nil
+		}
+
+		configConverter.Set("process_config.process_dd_url", processURLs[0])
+		for idx, url := range processURLs[1:] {
+			if url == "" || keys[idx] == "" {
```

Comment:
have you considered splitting these error cases into two errors -- one for  a missing url and one for an api_key-less URL ?